modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -1021,6 +1021,10 @@ class Session implements ISession {
         observers.each { trace -> trace.onFlowCreate(this) }
     }
 
+    void notifyFilePublish(Path destination) {
+        observers.each { trace -> trace.onFilePublish(destination) }
```

Comment:
Better to wrap this into a try-catch as done here 

https://github.com/nextflow-io/nextflow/pull/2523/files#diff-bee88a999f8eecd3cec5fcad24b76736162be86626f89d746932f64928f24374R997-R1004